econml/_ortho_learner.py

```diff
@@ -0,0 +1,584 @@
+                        sample_var=self._subinds_check_none(sample_var, fitted_inds))
+        return self
+
+    def _fit_nuisances(self, Y, T, X=None, W=None, Z=None, sample_weight=None):
+        # use a binary array to get stratified split in case of discrete treatment
+        splitter = check_cv(self._n_splits, [0], classifier=self._discrete_treatment)
+        # if check_cv produced a new KFold or StratifiedKFold object, we need to set shuffle and random_state
+        if splitter != self._n_splits and isinstance(splitter, (KFold, StratifiedKFold)):
+            splitter.shuffle = True
+            splitter.random_state = self._random_state
+
+        all_vars = [var if np.ndim(var) == 2 else var.reshape(-1, 1) for var in [Z, W, X] if var is not None]
+        if all_vars:
+            all_vars = np.hstack(all_vars)
+            folds = splitter.split(all_vars, T)
+        else:
+            folds = splitter.split(np.ones((T.shape[0], 1)), T)
+
+        if self._discrete_treatment:
+            T = self._label_encoder.fit_transform(T)
+            # drop first column since all columns sum to one
+            T = self._one_hot_encoder.fit_transform(reshape(T, (-1, 1)))[:, 1:]
+            self._d_t = shape(T)[1:]
+            self.transformer = FunctionTransformer(
+                func=(lambda T:
+                      self._one_hot_encoder.transform(
+                          reshape(self._label_encoder.transform(T), (-1, 1)))[:, 1:]),
+                validate=False)
+
+        nuisances, fitted_models, fitted_inds = _crossfit(self._model_nuisance, folds,
+                                                          Y, T, X=X, W=W, Z=Z, sample_weight=sample_weight)
+        self._models_nuisance = fitted_models
+        return nuisances, fitted_inds
+
+    def _fit_final(self, Y, T, X=None, W=None, Z=None, nuisances=None, sample_weight=None, sample_var=None):
+        self._model_final.fit(Y, T, **self._filter_none_kwargs(X=X, W=W, Z=Z,
+                                                               nuisances=nuisances, sample_weight=sample_weight,
+                                                               sample_var=sample_var))
+        self.score_ = None
+        if hasattr(self._model_final, 'score'):
+            self.score_ = self._model_final.score(Y, T, **self._filter_none_kwargs(X=X, W=W, Z=Z,
+                                                                                   nuisances=nuisances,
+                                                                                   sample_weight=sample_weight,
+                                                                                   sample_var=sample_var))
+
+    def const_marginal_effect(self, X=None):
+        self._check_fitted_dims(X)
+        if X is None:
+            return self._model_final.predict()
+        else:
+            return self._model_final.predict(X)
+    const_marginal_effect.__doc__ = LinearCateEstimator.const_marginal_effect.__doc__
```

Comment:
Perhaps we should create a decorator for this

Reply:
Hm. I'm fine without for now :) Then I need to remember we had this decorator :)